src/lib/viewers/controls/controls-layer/ControlsLayer.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import noop from 'lodash/noop';
+import './ControlsLayer.scss';
+
+export type Helpers = {
+    hide: () => void;
+    reset: () => void;
+    show: () => void;
+};
+
+export type Props = {
+    children: React.ReactNode;
+    onMount?: (helpers: Helpers) => void;
+    timeout?: number;
+};
+
+export const HIDE_CLASSNAME = 'bp-is-visible';
+export const HIDE_DELAY_MS = 2000;
+
+export default function ControlsLayer({ children, onMount = noop }: Props): JSX.Element {
+    const [isShown, setIsShown] = React.useState(false);
+    const hasFocusRef = React.useRef(false);
+    const hasCursorRef = React.useRef(false);
+    const hideTimeoutRef = React.useRef<number>();
+
+    // Visibility helpers
+    const helpers = React.useRef({
+        hide() {
+            window.clearTimeout(hideTimeoutRef.current);
+
+            hideTimeoutRef.current = window.setTimeout(() => {
```

Comment:
Are there any scenarios where we would want `hide` to immediately hide the control bar instead of waiting 2 seconds?

Reply:
Not currently, outside of the enable/disable methods in the root component. If it comes up at a later point, we can add more code to support it.